lib/cpp/src/thrift/transport/TSSLSocket.cpp

```diff
@@ -471,8 +471,10 @@ void TSSLSocket::checkHandshake() {
       }
     } while (rc == 2);
   } else {
-    // set the SNI hostname
-    SSL_set_tlsext_host_name(ssl_, getHost().c_str());
+    /* OpenSSL < 0.9.8f does not have SSL_set_tlsext_host_name() */
+    #if defined(SSL_set_tlsext_host_name) // set the SNI hostname
```

Comment:
I have created a issue regarding this: https://issues.apache.org/jira/browse/THRIFT-3892
No, there is not any alternative for older version. There are very less possibilities of someone using very old OpenSSL library. This issue is negligible, it is just for improving the project.